packages/SwingSet/src/kernel/kernel.js

```diff
@@ -684,12 +686,40 @@ export default function buildKernel(
     return harden(policyInput);
   }
 
+  async function processStartVat(message) {
+    postAbortActions.discardFailedDelivery = true;
+    const { type, vatID } = message;
+    // console.log(`-- processStartVat(${vatID})`);
+    insistVatID(vatID);
+    // eslint-disable-next-line no-use-before-define
+    if (!vatWarehouse.lookup(vatID)) {
+      // Note a bit of weirdness here: if a vat's module initialization or
+      // buildRootObject function throws an error, the crank in which it was
+      // executed will be aborted, which will leave the 'startVat' delivery on
```

Comment:
Wait, is that comment still true? I thought `discardFailedDelivery = true` means that the `start-vat`will be discarded. Should the `vatWarehouse` check be converted into an `assert(vatWarehouse.lookup(vatID))` instead?

Reply:
Yes, I think you're right.  I'll replace the comment with an assert of the contrary condition.